vignettes/ChromBand.Rmd

```diff
@@ -0,0 +1,379 @@
+
+</figure>
+
+For the analysis presented here, we consider the comparison of two subsets: patients identified as having a BCR/ABL gene fusion present, typically as a result of a translocation of chromosomes 9 and 22 (labeled BCR/ABL), and those that have no observed cytogenetic abnormalities (labeled NEG). The full dataset is available in the [ALL]{.sans-serif} package, and the relevant subset of the data can be obtained by
+
+```{r bcrAblOrNegSubset}
+data(ALL, package="ALL")
+subsetType <- "BCR/ABL"
+Bcell <- grep("^B", as.character(ALL$BT) )
+bcrAblOrNegIdx <- which(as.character(ALL$mol.biol) %in% c("NEG", subsetType) )
+bcrAblOrNeg <- ALL [, intersect(Bcell, bcrAblOrNegIdx) ]
+bcrAblOrNeg$mol.biol <- factor(bcrAblOrNeg$mol.biol)
+```
+
+We also create relevant annotation maps to go from feature names to Entrez ID, gene symbol, and chromosome band.
+
+```{r annMaps}
+annType <- c("db", "env")
+entrezMap <- getAnnMap("ENTREZID", annotation(bcrAblOrNeg),
+                       type=annType, load=TRUE)
+symbolMap <- getAnnMap("SYMBOL", annotation(bcrAblOrNeg),
+                       type=annType, load=TRUE)
+bandMap <- getAnnMap("MAP", annotation(bcrAblOrNeg),
+                     type=annType, load=TRUE)
+```
+
+We applied a non-specific filter to the dataset to remove probesets lacking the desired annotation as well as those with an interquartile range (IQR) below the median IQR, as probesets with little variation across samples are uninformative. We also ensured that each Entrez Gene identifier maps to exactly one probeset by selecting the probeset with the largest IQR when two or more probesets map to the same Entrez Gene ID.
+
+```{r nsFiltering}
+filterAns <- nsFilter(bcrAblOrNeg,
+                      require.entrez = TRUE,
+                      remove.dupEntrez = TRUE,
+                      var.func = IQR, var.cutoff = 0.5)
+nsFiltered <- filterAns$eset
+```
+
+We also remove probesets with no gene symbol, as well as those with no mapping to a chromosome band.
+
+```{r}
+hasSYM <- sapply(mget(featureNames(nsFiltered), symbolMap, ifnotfound=NA),
+                 function(x) length(x) > 0 && !is.na(x[1] ) )
+hasMAP <- sapply(mget(featureNames(nsFiltered), bandMap, ifnotfound=NA),
+                 function(x) length(x) > 0 && !is.na(x[1] ) )
+nsFiltered <- nsFiltered[hasSYM & hasMAP, ]
+```
+
+We define the *gene universe* to be the subset of genes that remain after this filtering.
+
+```{r defineGeneUniverse}
+affyUniverse <- featureNames(nsFiltered)
+entrezUniverse <- unlist(mget(affyUniverse, entrezMap))
+names(affyUniverse) <- entrezUniverse
+if (any(duplicated(entrezUniverse)))
+    stop("error in gene universe: can't have duplicate Entrez Gene Ids")
+```
+
+We assessed differential expression between the BCR/ABL and NEG groups using an empirical Bayes approach, as implemented in the software package limma [@limma], yielding an attenuated$t$-statistic for each gene.
```

Comment:
`limma` is also a Bioconductor package. Use

    `r Biocpkg("limma")`
    
to make the link.

Reply:
ok